From a4490c88c12c8746e42d78818d924a81cb0f4bec Mon Sep 17 00:00:00 2001
From: John Kirkham <kirkhamj@janelia.hhmi.org>
Date: Sun, 4 Mar 2018 02:53:55 -0500
Subject: [PATCH] Declare fsize at the beginning on main

Moves the `fsize` type definition to the top of `main` to comply with
C89's constraints. Should fix a compilation error on VS 2008 (needed for
Python 2.7 support on Windows).
---
 compat/filegen.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git compat/filegen.c compat/filegen.c
index 2ab4807..701ee62 100644
--- compat/filegen.c
+++ compat/filegen.c
@@ -34,6 +34,7 @@ int main(int argc, char *argv[]) {
   size_t osize = SIZE * sizeof(int32_t);
   int dsize = SIZE * sizeof(int32_t);
   int csize;
+  long fsize;
   int i;
 
   FILE *f;
@@ -77,7 +78,7 @@ int main(int argc, char *argv[]) {
     /* Read from argv[2] into data_out. */
     f = fopen(argv[2], "rb");
     fseek(f, 0, SEEK_END);
-    long fsize = ftell(f);
+    fsize = ftell(f);
     fseek(f, 0, SEEK_SET);
     if (fread(data_out, 1, (size_t) fsize, f) == fsize) {
       printf("Checking %s\n", argv[2]);
